backends/bmv2/backend.cpp

```diff
@@ -181,14 +181,16 @@ class RenameUserMetadata : public Transform {
 };
 
 void
-Backend::process(const IR::ToplevelBlock* tlb, BMV2Options& options) {
+Backend::process(const IR::ToplevelBlock* tlb, CompilerOptions& options) {
     CHECK_NULL(tlb);
     auto evaluator = new P4::EvaluatorPass(refMap, typeMap);
     if (tlb->getMain() == nullptr)
         return;  // no main
 
-    if (options.arch != Target::UNKNOWN)
-        target = options.arch;
+    if (options.arch == "v1model")
+        target = Target::SIMPLE;
+    else if (options.arch == "psa")
+        target = Target::PORTABLE;
```

Comment:
Nit: I would use the full names: `Target::SIMPLE_SWITCH` and `Target::PSA`  (or `Target::PORTABLE_SWITCH_ARCHITECTURE`)